### PR TITLE
refactor: remove dynamic scoping from browser module

### LINKED ISSUE #4246: [refactor] remove dynamic scoping from browser module

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
refactor: remove dynamic scoping from browser module

closes #4246

### What does the PR do

This PR aims at removing the dynamic scoping within the browser module. Whats done is -
1. Pass component from layout to the child components instead of them accessing it directly via dynamic scoping
2. Add and internal compose tin BrowserLayout so that accessing it from other components is not straightforward to discourage it in the future.
3. Move some logic to the store
4. Removed unnecessary import statements an cleaned up some code

### Affected areas

Browser

### Screenshot of functionality

https://user-images.githubusercontent.com/60327365/145122811-9d228ffe-5c32-4002-a2b1-ee9f9f34bda1.mov

### CODE REVIEW

**Comment by anastasiyaig** [comment]
@Khushboo-dev-cpp 

refreshing the page crashes the app

https://user-images.githubusercontent.com/82375995/145216252-cb3f923e-28c6-49cb-9b8d-f9e44680c9a5.mov

requesting STT does not open transaction modal in the dapp

```qrc:/app/AppLayouts/Browser/BrowserLayout.qml:153: ReferenceError: acc is not defined```

https://user-images.githubusercontent.com/82375995/145216491-9de21196-799b-41b1-bbbb-d009bac06fbb.mov

**Review by alexandraB99 — APPROVED** [review]
Great work! FYI ToastMessage is moved to Global [here](https://github.com/status-im/status-desktop/pull/4274), I will rebase as soon as this gets merged.

**Comment by anastasiyaig** [comment]
@Khushboo-dev-cpp i retested this again and noticed that validation when signing transaction in browser is no longer present, so I can sign transaction with wrong password. Is it something got broken here or it is already present in base_bc?

**Comment by Khushboo-dev-cpp** [comment]
@anastasiyaig thanks for noticing. I just checked and it exists on base_bc generally. Maybe we can raise a separate bug for it?

**Comment by anastasiyaig** [comment]
@Khushboo-dev-cpp ya, i will test base_bc separately then and we will fix it there :)

**Review by anastasiyaig — APPROVED** [review]

File: ui/app/AppLayouts/Browser/BrowserLayout.qml [hunk 1]
```diff
@@ -22,115 +22,119 @@ import "../Chat/popups"
...
-    property Component sendTransactionModalComponent: SignTransactionModal {
-        store: browserWindow.globalStore
-    }
+        property Component jsDialogComponent: JSDialogWindow {}
 
-    property Component signMessageModalComponent: SignMessageModal {}
+        property Component accessDialogComponent: BrowserConnectionModal {
+            currentTab: tabs.getTab(tabs.currentIndex) && tabs.getTab(tabs.currentIndex).item
+            // the 78 is for width of the navBar
+            x: browserWindow.width - width - Style.current.halfPadding + 78
```
Comment (jrainville) [inline ui/app/AppLayouts/Browser/BrowserLayout.qml:49]: We might want to use a constant or a dynamic value here, in case the navBar changes someday

**Review by jrainville — COMMENTED** [review]
Nice job.

I see that there is a lot of remaining TODOs in BrowserLayout about moving stuff to stores or Global. Is that something that will be addressed by another PR, or maybe we need to keep track of it in anther issue?

Also, BrowserLayout was already pretty massive, so maybe in another PR someday we should split it up a little.

File: ui/app/AppLayouts/Browser/BrowserLayout.qml (hunk 1 above)
Comment (Khushboo-dev-cpp) [reply ui/app/AppLayouts/Browser/BrowserLayout.qml]: Found a better way to do it :) thanks

**Comment by Khushboo-dev-cpp** [comment]
@anastasiyaig thanks a much for identifying these. Phew. Will fix them asap


> Nice job.
> 
> I see that there is a lot of remaining TODOs in BrowserLayout about moving stuff to stores or Global. Is that something that will be addressed by another PR, or maybe we need to keep track of it in anther issue?
> 
> Also, BrowserLayout was already pretty massive, so maybe in another PR someday we should split it up a little.

Hey so again thanks for pointing out. ON his based on a PR Alex is currently working on and  I also added that to the note with the Todo.

Got rid of 2 of them, the rest were already there, not touching them for now.

**Review by jrainville — APPROVED** [review]

**Pull request merged**